acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/dsl/signer/EthSignerProcessRunner.java

```diff
@@ -140,11 +139,25 @@ public void start(final String processName) {
     params.add(String.valueOf(signerHttpRpcPort));
     params.add("--chain-id");
     params.add(chainId);
-
     if (useDynamicPortAllocation) {
       params.add("--data-directory");
       params.add(dataDirectory.toAbsolutePath().toString());
     }
+    if (hashicorpVaultPort == 0) {
```

Comment:
It would be nicer to have an object passed in that created the signer portion of this string (e.g. CommandPattern style) ...as then this `EthSignerProcessRunner` wouldn't have to change on every additional integration

Reply:
I think this process runner could be improved. When we remove all that logic that is in there to create the ethsigner process it would look a lot nicer. All that logic could be moved into the SignerConfiguration. SignerConfiguration then would need the NodeConfiguration and the NodePorts.
That means if there are changes in configuration the process runner would not be affected anymore.
What do you think?

Reply:
Design is unclear, but I don't like the sound of moving logic into a data structure object.